src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.hmcts.reform.sendletter.services;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sendletter.cache.SentLettersCache;
+import uk.gov.hmcts.reform.sendletter.exceptions.LetterAlreadySentException;
+import uk.gov.hmcts.reform.sendletter.model.Letter;
+import uk.gov.hmcts.reform.sendletter.notify.INotifyClient;
+
+@Service
+public class LetterService {
+
+    private final SentLettersCache cache;
+    private final INotifyClient notifyClient;
+
+    public LetterService(
+        SentLettersCache cache,
+        INotifyClient notifyClient
+    ) {
+        this.cache = cache;
+        this.notifyClient = notifyClient;
+    }
+
+    public void send(Letter letter) {
+        if (cache.add(letter)) {
+            try {
+                notifyClient.send(letter);
```

Comment:
Could we simplify this as below ? Exception would be handled by Controller if the notify client fails
```
    public void send(Letter letter){
     	if(!cache.add(letter)){
     		throw new LetterAlreadySentException("Duplicate request.Letter already sent.");
     	}

     	notifyClient.send(letter);
     	cache.add(letter);
     }

```

Reply:
I don't find it simpler. Also not that the code you've pasted above adds letter to cache twice.

Reply:
I thought `cache.add(letter)` will not add if the letter is already present and return false.Sorry for the confusion

Reply:
that's what i mean, if it returns `true` it's called twice in your code.

Reply:
looks good thanks for clarifying